### PR TITLE
Changing config without restarting

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
Hi,

This PR contains multiple commit. Only the last one is important: it ensures that you don't need to restart in order to use new configuration values.

The other one are just some cleaning using pep8 and sorting of imports

### CODE REVIEW

**Comment by Arthur-Milchior** [comment]
I don't know whether it's something normal, but the code on github is not the same as the version currently on ankiweb

File: __init__.py [hunk 1]
```diff
@@ -2,32 +2,32 @@
...
 
-
-config = mw.addonManager.getConfig(__name__)
-
 ankiver_minor = int(ankiversion.split('.')[2])
 ankiver_major = ankiversion[0:3]
 
-#Get js files.
+# Get js files.
+
```
Comment (BlueGreenMagick) [inline __init__.py]: There shouldn't be any blank lines here, as the above comment describes the function below

File: __init__.py [hunk 2]
```diff
@@ -36,52 +36,63 @@ def js_from_path(path):
...
 def bool_to_str(b):
     if b:
         return "true"
     else:
         return ""
 
-#Code for new style hooks.
+# Code for new style hooks.
+
```
Comment (BlueGreenMagick) [inline __init__.py]: These blank lines should be removed as well.

**Comment by BlueGreenMagick** [comment]
The add-on on Ankiweb is currently v3.3. I was planning to upload a new version when I was pretty sure there wasn't any critical bugs in it.

**Comment by BlueGreenMagick** [comment]
I've went through your changes to how the addon handles config, and I'm not sure it is better to change config without restarting. If for example one changes `ctrl_click` config from `false` to `true` while reviewing, wouldn't `GLOBALCARDJS` think `ctrl` is false while CARDJS think `ctrl` is true? The changes wouldn't take effect anyway.

**Comment by Arthur-Milchior** [comment]
Most add-on I know re-create the JS each time the web editor is created. I guess I have not realized that here, the js was a constant. However, that's quite easy to fix

**Comment by BlueGreenMagick** [comment]
I do indeed see the benefits of having a config that does not require a restart. I think when config is modified, if we check if state is reviewer and if so, re-initialize the web, there shouldn't be a problem.

I've also got a few more questions about some of the config functions, of which purpose I'm not quite sure I understand.

File: config.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    userOption = None
+    fromName = None
+
+
+mw.addonManager.setConfigUpdatedAction(__name__, update)
+
+fromName = None
+
+
+def getFromName(name):
```
Comment (BlueGreenMagick) [inline config.py]: What is the purpose of this function? This does not seem to get called from anywhere

File: config.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    global userOption
+    if userOption is None:
+        userOption = mw.addonManager.getConfig(__name__)
+
+
+def getUserOption(key=None, default=None):
+    """Get the user option if it is set. Otherwise return the default
+    value and add it to the config.
+
+    When an add-on was updated, new config key were not added. This
```
Comment (BlueGreenMagick) [inline config.py]: Could I ask what you mean by it? Maybe I misunderstood, but I tested with a new addon with 'abc' in its config, then created a new version to also have 'bcd' in the config. I uploaded and downloaded the first version in Anki, then upload-update to the second version. And I could see both 'abc' and 'bcd' in the addon config editor in Anki.

File: config.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+
+# Copyright GPL3.
+# Code by Arthur Milchior <arthur@milchior.fr>
+
+import sys
+
+from aqt import mw
+from aqt.utils import showWarning
```
Comment (BlueGreenMagick) [inline config.py]: This import seems redundant

**Comment by BlueGreenMagick** [comment]
Sorry, but after some thoughts, due to the way this addon works. I think it will be much simpler if the copy of config was updated each time Review.revHTML is called, instead of listening to config changes. I merged the first 2 commits and discarded the last one.

**Pull request closed without merging**